### DISCUSSION TITLE
Get the value of a TextColumn

### DISCUSSION THREAD

**Discussion by DarkCoderJ** [filamentphp/filament#5604] [Help] [answered]
I have this TextColumn
` TextColumn::make('partido')->sortable(),`

Its value is for example 45, but it can be diferent in every row and i want to get this value to use it in other TextColumn:

`TextColumn::make('liga')->default(Liga::where('id', the value of 'TextColumn::make('partido')),`

Is it possible to get that value?

Thanks

**Comment by zhanang19** [comment]
You can use insert callback for this functionality, your code will look like this:

```php
TextColumn::make('liga')
    ->default(fn ($record): ?string => $record->partido) // $record->partido is the value of TextColumn::make('liga')
```

**Reply by zepfietje** [reply]
This isn't correct as it doesn't get the current value of the input.

**Reply by zhanang19** [reply]
Its correct if the the `TextColumn::make('liga')` value is not modified (default behaviour).

Its incorrect if the `TextColumn::make('liga')` value is already manipulated using  `getStateUsing`.

please CMIIW.

**Reply by zepfietje** [reply]
I misread and misunderstood the discussion, thought it was about inputs.
I've updated the marked answer.

**Reply by DarkCoderJ** [reply]
Thanks for your answer. That works this way:
`TextColumn::make('liga')
    ->default(fn ($record): ?string => $record->partido)`
but it doesn't work if try to use it in a query like this
`TextColumn::make('liga')->default(Resultado::where('partido',fn ($record): ?string => $record->partido)->implode('partido'))`
I get this error message: Undefined property: Illuminate\Database\Query\Builder::$partido

Thanks

**Reply by danharrin** [reply]
The callback always needs to be inside the default(), you can't just add it inside the where()

**Reply by DarkCoderJ** [reply]
Ok, thank you. I found the way to make it work:
`TextColumn::make('liga')->default(fn (Resultado $record): ?string => Liga::where('id', Partido::where('id', $record->partido)->implode('Liga_id'))->implode('nombre')),`

**Reply by danharrin** [reply]
This definitely isn't very performant and will create an N+1 issue. You should define an eloquent relationship for this

**Comment by zepfietje** [comment, marked as answer]
https://filamentphp.com/docs/2.x/tables/columns/getting-started#calculated-state

**Reply by zhanang19** [reply]
I think what the OP ask is about `TextColumn` component, not a `TextInput` component

Please CMIIW @zepfietje @DarkCoderJ

**Reply by zepfietje** [reply]
Oh sorry, I misread. Have updated the link.
